### PR TITLE
Issue 707 - Remove Ecto Model references

### PR DESCRIPTION
**Pull request by mcelaney**
This PR is a work in progress.

Working to resolve Issue 707 (https://github.com/phoenixframework/phoenix_guides/issues/707) - removing references to Ecto Models.

- [x] introduction overview
-  testing models <- handled in https://github.com/phoenixframework/phoenix_guides/issues/721
- [x] mix tasks
- [x] seeding data
- [x] custom primary key
- testing controllers <- handled in https://github.com/phoenixframework/phoenix_guides/issues/721
- [x] file uploads
- [x] channels
- [x] Other...

### CODE REVIEW

File: docs/bonus_guides/mix_tasks.md [hunk 1]
```diff
@@ -193,109 +197,159 @@ defmodule Hello.Mixfile do
...
-$ mix phoenix.gen.html Post posts body:string word_count:integer --no-model
-* creating web/controllers/post_controller.ex
-* creating web/templates/post/edit.html.eex
-* creating web/templates/post/form.html.eex
-* creating web/templates/post/index.html.eex
-* creating web/templates/post/new.html.eex
-* creating web/templates/post/show.html.eex
-* creating web/views/post_view.ex
-* creating test/controllers/post_controller_test.exs
+$ mix phx.gen.html Flog Post posts body:string word_count:integer --no-context
```
Comment (Gazler) [inline docs/bonus_guides/mix_tasks.md]: s/Flog/Blog

File: docs/bonus_guides/mix_tasks.md [hunk 2]
```diff
@@ -193,109 +197,159 @@ defmodule Hello.Mixfile do
...
+** (CompileError) lib/hello_web/templates/post/edit.html.eex:3: undefined function post_path/3
+    (stdlib) lists.erl:1338: :lists.foreach/2
+    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6
+    (elixir) lib/kernel/parallel_compiler.ex:121: anonymous fn/4 in Kernel.ParallelCompiler.spawn_compilers/1
 ```
 
 #### `mix phx.gen.json`
 
-Phoenix also offers the ability to generate all the code to stand up a complete JSON resource - ecto migration, ecto model, controller with all the necessary actions and view. This command will not create any template for the app.
+Phoenix also offers the ability to generate all the code to stand up a complete JSON resource - ecto migration, ecto context, controller with all the necessary actions and view. This command will not create any template for the app.
```
Comment (Gazler) [inline docs/bonus_guides/mix_tasks.md]: s/context/schema

File: docs/channels.md [hunk 3]
```diff
@@ -234,13 +234,13 @@ end
...
 #### Intercepting Outgoing Events
-We won't implement this for our application, but imagine our chat app allowed users to ignore messages about new users joining a room. We could implement that behavior like this where we explicitly tell Phoenix which outgoing event we want to intercept and then define a `handle_out/3` callback for those events. (Of course, this assumes that we have a `User` model with an `ignoring?/2` function, and that we pass a user in via the `assigns` map.)
+We won't implement this for our application, but imagine our chat app allowed users to ignore messages about new users joining a room. We could implement that behavior like this where we explicitly tell Phoenix which outgoing event we want to intercept and then define a `handle_out/3` callback for those events. (Of course, this assumes that we have a `Accounts` context with an `ignoring?/2` function, and that we pass a user in via the `assigns` map.)
 
 ```elixir
 intercept ["user_joined"]
 
 def handle_out("user_joined", msg, socket) do
-  if User.ignoring?(socket.assigns[:user], msg.user_id) do
+  if Accounts.ignoring?(socket.assigns[:user], msg.user_id) do
```
Comment (Gazler) [inline docs/channels.md]: Should we use Accounts.igoring_user?

File: docs/bonus_guides/custom_primary_key.md [hunk 4]
```diff
@@ -13,20 +13,24 @@ Let's say that we need a JSON resource that stores rows of team athletes. Each a
...
+* creating lib/hello/controllers/fallback_controller.ex
+* creating lib/hello/rosters/player.ex
+* creating priv/repo/migrations/20170906132454_create_players.exs
+* creating lib/hello/rosters/rosters.ex
+* injecting lib/hello/rosters/rosters.ex
+* creating test/hello/rosters/rosters_test.exs
+* injecting test/hello/rosters/rosters_test.exs
 
-Add the resource to your api scope in web/router.ex:
+Add the resource to your :api scope in lib/hello/router.ex:
```
Comment (jeregrine) [inline docs/bonus_guides/custom_primary_key.md]: I don't think the symbol notation is necessary. BUT if you do have strong feelings for :api, put it in code blocks `:api`

File: docs/bonus_guides/custom_primary_key.md [hunk 5]
```diff
@@ -13,20 +13,24 @@ Let's say that we need a JSON resource that stores rows of team athletes. Each a
...
+* creating lib/hello/views/player_view.ex
+* creating test/hello/controllers/player_controller_test.exs
+* creating lib/hello/views/changeset_view.ex
+* creating lib/hello/controllers/fallback_controller.ex
+* creating lib/hello/rosters/player.ex
+* creating priv/repo/migrations/20170906132454_create_players.exs
+* creating lib/hello/rosters/rosters.ex
+* injecting lib/hello/rosters/rosters.ex
+* creating test/hello/rosters/rosters_test.exs
+* injecting test/hello/rosters/rosters_test.exs
```
Comment (nekath) [inline docs/bonus_guides/custom_primary_key.md]: Web related files are in hello_web directory :)
```
Compiling 4 files (.ex)
Generated phoenix_ecto app
==> hello
* creating lib/hello_web/controllers/player_controller.ex
* creating lib/hello_web/views/player_view.ex
* creating test/hello_web/controllers/player_controller_test.exs
* creating lib/hello_web/views/changeset_view.ex
* creating lib/hello_web/controllers/fallback_controller.ex
* creating lib/hello/rosters/player.ex
* creating priv/repo/migrations/20170906192146_create_players.exs
* creating lib/hello/rosters/rosters.ex
* injecting lib/hello/rosters/rosters.ex
* creating test/hello/rosters/rosters_test.exs
* injecting test/hello/rosters/rosters_test.exs

Add the resource to your :api scope in lib/hello_web/router.ex:
```

File: docs/bonus_guides/custom_primary_key.md (hunk 4 above)
Comment (nekath) [inline docs/bonus_guides/custom_primary_key.md]: `lib/hello_web/router.ex` :)

File: docs/bonus_guides/custom_primary_key.md [hunk 6]
```diff
@@ -77,14 +84,15 @@ defmodule Hello.Player do
 
     timestamps
   end
+
   . . .
 ```
 
-There's just one more thing we'll need to do, and that's remove the reference to `id: player.id,` in the `def render("player.json", %{player: player})` function body.
+There's just one more thing we'll need to do, and that's remove the reference to `id: player.id,` in the `def render("player.json", %{player: player})` function body in `lib/hello/views/player_view.ex`.
```
Comment (nekath) [inline docs/bonus_guides/custom_primary_key.md]: `lib/hello_web/views/player_view.ex`

File: docs/bonus_guides/custom_primary_key.md (hunk 4 above)
Comment (nekath) [reply docs/bonus_guides/custom_primary_key.md]: @jeregrine This is an output of above `mix phx.gen.json ... ` command.

File: docs/introduction/installation.md [hunk 7]
```diff
@@ -87,7 +87,7 @@ $ ln -s /usr/bin/nodejs /usr/bin/node
 
 PostgreSQL is a relational database server. Phoenix configures applications to use it by default, but we can switch to MySQL by passing the `--database mysql` flag when creating a new application.
 
-When we work with Ecto models in these guides, we will use PostgreSQL and the Postgrex adapter for it. In order to follow along with the examples, we should install PostgreSQL. The PostgreSQL wiki has [installation guides](https://wiki.postgresql.org/wiki/Detailed_installation_guides) for a number of different systems.
+When we work with Ecto contexts in these guides, we will use PostgreSQL and the Postgrex adapter for it. In order to follow along with the examples, we should install PostgreSQL. The PostgreSQL wiki has [installation guides](https://wiki.postgresql.org/wiki/Detailed_installation_guides) for a number of different systems.
```
Comment (nekath) [inline docs/introduction/installation.md]: Shouldn't be schemas ?

**Comment by Gazler** [comment]
Thanks <3 <3 <3

**Pull request merged**